src/Telegram.Bot/Requests/FileRequestBase.cs

```diff
@@ -0,0 +1,80 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http;
+using Newtonsoft.Json;
+using Telegram.Bot.Helpers;
+using Telegram.Bot.Types;
+
+namespace Telegram.Bot.Requests
+{
+
+    /// <summary>
+    /// Represents an API request with a file
+    /// </summary>
+    /// <typeparam name="TResponse">Type of result expected in result</typeparam>
+    public abstract class FileRequestBase<TResponse> : RequestBase<TResponse>
+    {
+        /// <summary>
+        /// Initializes an instance of request
+        /// </summary>
+        /// <param name="methodName">Bot API method</param>
+        protected FileRequestBase(string methodName)
+            : base(methodName)
+        { }
+
+        /// <summary>
+        /// Initializes an instance of request
+        /// </summary>
+        /// <param name="methodName">Bot API method</param>
+        /// <param name="method">HTTP method to use</param>
+        protected FileRequestBase(string methodName, HttpMethod method)
+            : base(methodName, method)
+        { }
```

Comment:
Not sure whether this ctor will be needed. AFAIK, all file sending requests are POST